src/data/tabular.jl

```diff
@@ -111,3 +111,54 @@ function load_credit_default(n::Union{Nothing,Int}=5000)
 
     return counterfactual_data
 end
+
+"""
+    load_german_credit(n::Union{Nothing, Int}=nothing)
+
+Loads and pre-processes UCI German Credit data.
+
+# Arguments
+- `n::Union{Nothing, Int}=nothing`: The number of samples to subsample from the dataset. If `n` is not specified, all samples will be used. Must be <= 1000 and >= 1.
+
+# Returns
+- `counterfactual_data::CounterfactualData`: A `CounterfactualData` object containing the preprocessed data.
+
+# Example
+data = load_german_credit(500) # loads and preprocesses 500 samples from the German Credit dataset
+
+"""
```

Comment:
I like this way of clearly specifying the arguments and the return value and giving an example - we should use this kind of documentation in other PRs as well.